packages/config/src/main.js

```diff
@@ -197,6 +205,11 @@ const mergeAndNormalizeConfig = function ({ config, defaultConfig, inlineConfig,
   const configC = mergeContext(configB, context, branch)
 
   validatePreNormalizeConfig(configC)
+
+  if (isFirstPass) {
+    warnLegacyFunctionsDirectory({ config: configC, logs })
```

Comment:
[sand] What are your thoughts on adding a `functionsDirectoryOrigin: 'configLegacy'` value?
This would allow calling `warnLegacyFunctionsDirectory()` outside of `mergeAndNormalizeConfig()`, to avoid using `isFirstPass`.

Reply:
YES! I hated `isFirstPass` and this feels like a great alternative. Thanks! 🙌🏻

Reply:
I have done this in https://github.com/netlify/build/pull/2496/commits/ab51561f3b850b5c0f9971b7b55b9bbf3333135e. I've used `config-v1` instead of `configLegacy`, because if we ever deprecate the current syntax, we can call it `config-v2`, etc.

Reply:
This looks great!